common/persistence/sql/queue_v2.go

```diff
@@ -26,36 +26,183 @@ package sql
-	return nil, fmt.Errorf("%w: EnqueueMessage", ErrNotImplemented)
+	var lastMessageID int64
+	tx, err := q.Db.BeginTx(ctx)
+	if err != nil {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("EnqueueMessage failed. Failed to start transaction. Error: %v", err))
+	}
+	lastMessageID, err = tx.GetLastEnqueuedMessageIDForUpdateV2(ctx, sqlplugin.QueueV2Filter{
+		QueueType: request.QueueType,
+		QueueName: request.QueueName,
+		Partition: EmptyPartition,
+	})
+	switch err {
+	case nil:
+		lastMessageID = lastMessageID + 1
+	case sql.ErrNoRows:
+		lastMessageID = persistence.FirstQueueMessageID
+	default:
+		rollBackErr := tx.Rollback()
+		if rollBackErr != nil {
+			q.logger.Error("transaction rollback error", tag.Error(rollBackErr))
+		}
+		return nil, fmt.Errorf("failed to get last enqueued message id: %w", err)
+	}
+	_, err = tx.InsertIntoQueueV2Messages(ctx, []sqlplugin.QueueV2MessageRow{
+		newQueueV2Row(request.QueueType, request.QueueName, lastMessageID, request.Blob),
+	})
+	if err != nil {
+		rollBackErr := tx.Rollback()
+		if rollBackErr != nil {
+			q.logger.Error("transaction rollback error", tag.Error(rollBackErr))
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("EnqueueMessage failed. Failed to commit transaction. Error: %v", err))
+	}
+	return &persistence.InternalEnqueueMessageResponse{Metadata: persistence.MessageMetadata{ID: lastMessageID}}, err
 }
 
-func (q queueV2) ReadMessages(
-	context.Context,
-	*persistence.InternalReadMessagesRequest,
+func (q *queueV2) ReadMessages(
+	ctx context.Context,
+	request *persistence.InternalReadMessagesRequest,
 ) (*persistence.InternalReadMessagesResponse, error) {
-	return nil, fmt.Errorf("%w: ReadMessages", ErrNotImplemented)
+	if request.PageSize <= 0 {
+		return nil, persistence.ErrNonPositiveReadQueueMessagesPageSize
+	}
+	var minMessageID int64
+	minMessageID, err := q.getMinMessageID(request)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := q.Db.RangeSelectFromQueueV2Messages(ctx, sqlplugin.QueueV2MessagesFilter{
+		QueueType:    request.QueueType,
+		QueueName:    request.QueueName,
+		Partition:    EmptyPartition,
+		MinMessageID: minMessageID,
+		PageSize:     request.PageSize,
+	})
+
+	if err != nil {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("RangeSelectFromQueueV2Messages operation failed. Error %v", err))
+	}
+
+	var messages []persistence.QueueV2Message
+	for _, row := range rows {
+		messages = append(messages, persistence.QueueV2Message{
+			MetaData: persistence.MessageMetadata{ID: row.MessageID},
+			Data:     *persistence.NewDataBlob(row.MessagePayload, row.MessageEncoding),
+		})
+	}
+	nextPageToken := q.getNextPageToken(messages)
+
+	response := &persistence.InternalReadMessagesResponse{
+		Messages:      messages,
+		NextPageToken: nextPageToken,
+	}
+	return response, nil
+}
+
+func (q *queueV2) getNextPageToken(result []persistence.QueueV2Message) []byte {
```

Comment:
I have moved this code to common/persistence. But Cassandra implementation differs a little bit now because of reading minMessageId from the metadata table. In the next PR, I will make both of them similar and remove that code from Cassandra's queue_v2_store.go